modules/common/test/helpers/deployment.go

```diff
@@ -50,6 +50,24 @@ func (tc *TestHelper) GetDeployment(name types.NamespacedName) *appsv1.Deploymen
 	return deployment
 }
 
+// SimulateDeploymentAnyNumberReplicaReady function retrieves the Deployment resource and
+// simulate that replicas are ready
+// Example usage:
+//
+//	th.SimulateDeploymentAnyNumberReplicaReady(ironicNames.INAName, 0)
+func (tc *TestHelper) SimulateDeploymentAnyNumberReplicaReady(name types.NamespacedName, replica int32) {
+	gomega.Eventually(func(g gomega.Gomega) {
+		deployment := tc.GetDeployment(name)
+
+		deployment.Status.Replicas = replica
```

Comment:
could you please also change SimulateDeploymentReplicaReady() to set Status.Replicas and ReadyReplicas based on deployment.Replicas to allow the positive test cases to still use the SimulateDeploymentReplicaReady call an only the negative cases need to use the new SimulateDeploymentAnyNumberReplicaReady

Reply:
done, should be ok also with other operators